html_xblock/static/js/html.js

```diff
@@ -0,0 +1,267 @@
+const FONTS = CUSTOM_FONTS + STANDARD_FONTS;
+
+function openTab(evt, tabName) {
+  /**
+   * This method has been adopted as found in https://www.w3schools.com/howto/howto_js_tabs.asp
+   */
+  var i, tabcontent, tablinks;
+  tabcontent = document.getElementsByClassName("tabcontent");
+  for (i = 0; i < tabcontent.length; i++) {
+    tabcontent[i].style.display = "none";
+  }
+  tablinks = document.getElementsByClassName("tablinks");
+  for (i = 0; i < tablinks.length; i++) {
+    tablinks[i].className = tablinks[i].className.replace(" active", "");
+  }
+  document.getElementById(tabName).style.display = "block";
+  evt.currentTarget.className += " active";
+}
+
+function configureTheEditor(data) {
+  const contentSelector = "textarea#html5-textarea";
+  const languageWrapper = document.querySelectorAll(".wrapper-view, .window-wrap");
+  const directionality = (languageWrapper.length > 0) ? languageWrapper.dir : "ltr";
+  var editor;
+
+  if (data.editor === "visual") {
+    tinymce.remove(contentSelector);
+    editor = tinymce.init({
+      skin_url: data.skin_url,
+      theme: "modern",
+      schema: "html5",
+      convert_urls: false,
+      directionality: directionality,
+      selector: contentSelector,
+      menubar: false,
+      statusbar: false,
+      valid_elements: "*[*]",
+      extended_valid_elements: "*[*]",
+      valid_children: "+body[style]",
+      invalid_elements: "",
+      font_formats: FONTS,
+      toolbar: "formatselect | fontselect | bold italic underline forecolor codesample | alignleft aligncenter alignright alignjustify | bullist numlist outdent indent blockquote | link unlink image | code",
+      external_plugins: data.external_plugins,
+      formats: {
+        code: {
+          inline: 'code'
+        }
+      },
+      visual: false,
+      image_advtab: true,
+      block_formats: "Paragraph=p;Preformatted=pre;Heading 3=h3;Heading 4=h4;Heading 5=h5;Heading 6=h6",
+      width: '100%',
+      height: '315px',
+      browser_spellcheck: true,
+      codemirror: {
+        width: 770,
+        height: 454,
+        saveCursorPosition: true,
+        config: {
+          mode: 'text/html',
+        }
+      }
+    });
+  } else {
+    editor = CodeMirror.fromTextArea(document.querySelectorAll(contentSelector)[0], {
+      mode: "text/html",
+      lineNumbers: true,
+      matchBrackets: true,
+      lineWrapping: true,
+    });
+  }
+
+  return editor;
+}
+
+function getSettingsValues(fields) {
+  var values = {};
+  var notSet = []; // List of field names that should be set to default values
+  for (var i in fields) {
+    var field = fields[i];
+    if (field.isSet()) {
+      values[field.name] = field.val();
+    } else {
+      notSet.push(field.name);
+    }
+  }
+
+  return {
+    values: values,
+    defaults: notSet
+  }
+}
+
+function extractXBlockFields() {
+  /**
+   * The content of this function is as found in xblockutils#studio_edit.js
```

Comment:
Is this code really necessary? A lot of this is for auto-generating the editor fields, but in this case it seems we could just hard-code the field HTML since there are only a couple of settings.